payments/models.py

```diff
@@ -1109,9 +1109,78 @@ def clean(self):
             raise ValidationError(_("The application has to be a switch application"))
 
         # Validate that the lease can only be from the immediate last season
-        if self.lease.start_date.year != calculate_season_start_date(
-            today()
-        ) - relativedelta(years=1):
+        if (
+            self.lease.start_date.year
+            != (calculate_season_start_date(today()) - relativedelta(years=1)).year
+        ):
             raise ValidationError(
                 _("The exchanged lease has to be from the last season")
             )
+
+    def set_status(self, new_status: OfferStatus, comment: str = None) -> None:
```

Comment:
+1 Actually I'll edit the offer expiration cronjob to use this instead of settings status directly